tests/integration/targets/win_dns_zone/tasks/activedirectory.yml

```diff
@@ -0,0 +1,306 @@
+---
+- name: Ensure AD/DNS roles are installed
+  ansible.windows.win_feature:
+    name:
+      - AD-Domain-Services
+      - DNS
+    include_management_tools: true
+    include_sub_features: true
+    state: present
+
+- name: Ensure domain is present
+  ansible.windows.win_domain:
```

Comment:
We should try and use `microsoft.ad.domain` module here. I'll try and submit a PR to add it as a test requirement as it is somewhat complicated to get it set up.

Reply:
https://github.com/ansible-collections/ansible.windows/pull/691 has been merged so now `microsoft.ad` is available in the integration tests.

Reply:
Done.